src/lxml/tests/test_xmlschema.py

```diff
@@ -80,7 +80,7 @@ def test_xmlschema_error_log_path(self):
         schema.validate(tree)
         tree_path = tree.getpath(tree.findall('b')[1])
         error_path = schema.error_log[0].path
-        self.assertTrue(tree_path == error_path)
+        self.assertTrue(error_path is None or tree_path == error_path)
```

Comment:
I went ahead and rolled back to the tighter version of the test, with a comment explaining what to do if our assumption about always having a location for schema validation errors is no longer valid at some point in the future.